File: src/lib/ts/helpers/discord.ts
import { formattedNow } from '$ts/helpers/time';

export function getDiscordWebhookBodyNewSubscriber({
	plan,
	email,
	supabaseId,
	stripeId,
	count
}: {
	plan: string;
	email: string;
	supabaseId: string;
	stripeId: string;
	count: number | null;
}) {
	return {
		content: null,
		embeds: [
			{
				title: `🎉 New Subscriber${count !== null ? ` #${count}` : ''}`,
				color: 11437567,
				fields: [
					{
						name: 'Email',
						value: email
					},
					{
						name: 'Plan',
						value: plan
					},
					{
						name: 'Supabase ID',
						value: supabaseId
					},
					{
						name: 'Stripe ID',
						value: stripeId
					}
				],
				footer: {
					text: formattedNow()
				}
			}
		],
		attachments: []
	};
}

File: go-apps/cron/models/discord.go
package models

type DiscordWebhookBody struct {
	Content     *string                    `json:"content"`
	Embeds      []DiscordWebhookEmbed      `json:"embeds"`
	Attachments []DiscordWebhookAttachment `json:"attachments"`
}

type DiscordWebhookEmbed struct {
	Title  string                    `json:"title,omitempty"`
	Fields []DiscordWebhookField     `json:"fields"`
	Color  int                       `json:"color"`
	Footer DiscordWebhookEmbedFooter `json:"footer"`
}

type DiscordWebhookEmbedFooter struct {
	Text string `json:"text"`
}

type DiscordWebhookField struct {
	Name   string `json:"name"`
	Value  string `json:"value"`
	Inline bool   `json:"inline"`
}

type DiscordWebhookAttachment struct {
	Filename string `json:"filename"`
	Content  string `json:"content"`
}
